File: 2023/tests/day14.test.js
const fs = require('fs');
const { part1, part2 } = require('../src/day14');

const example1 = `
O....#....
O.OO#....#
.....##...
OO.#O....O
.O.....O#.
O.#..O.#.#
..O..#O..O
.......O..
#....###..
#OO..#....
`.trim();

const input = fs.readFileSync('inputs/input14.txt', 'utf8');

describe('part 1', () => {
  test('example', () => {
    expect(part1(example1)).toBe(136);
  });

  test('solution', () => {
    expect(part1(input)).toBe(109596);
  });
});

describe('part 2', () => {
  test('example', () => {
    expect(part2(example1)).toBe(64);
  });

  test('solution', () => {
    expect(part2(input)).toBe(96105);
  });
});

File: 2024/AdventOfCode/Tests/Day14Test.cs
using NUnit.Framework;

namespace AdventOfCode.Tests;

[TestFixture]
public class Day14Test {
    private const string Example = "Inputs/14-Example.txt";
    private const string Input = "Inputs/14.txt";
    private static Day14 _day;

    [TestFixture]
    public class Part1 {
        [Test]
        public void TestExample() {
            _day = new Day14(Example);
            const string expected = "12";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day14(Input);
            const string expected = "208437768";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }

    [TestFixture]
    public class Part2 {
        [Test]
        public void TestInput() {
            _day = new Day14(Input);
            const string expected = "7492";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }
}